tests/pass/tls/macos_tlv_atexit.rs

```diff
@@ -1,34 +1,40 @@
 //@only-target-darwin
 
-use std::ptr;
 use std::thread;
 
 extern "C" {
     fn _tlv_atexit(dtor: unsafe extern "C" fn(*mut u8), arg: *mut u8);
 }
 
-fn register(dtor: extern "C" fn(*mut u8)) {
-    unsafe {
-        _tlv_atexit(dtor, ptr::null_mut());
+fn register<F>(f: F)
+where
+    F: FnOnce() + 'static,
+{
+    unsafe extern "C" fn run<F>(data: *mut u8)
+    where
+        F: FnOnce() + 'static,
+    {
+        unsafe { (*Box::from_raw(data as *mut F))() }
```

Comment:
It took me a bit to realize you are also calling the function here. Please let-bind the `Box::from_raw(data as *mut F)` so that this becomes more clear.